tests/Unit/VisitTransfer/ReferenceTest.php

```diff
@@ -64,4 +66,37 @@ public function itCanRejectItself()
         $reference->reject();
         $this->assertEquals(Reference::STATUS_REJECTED, $reference->fresh()->status);
     }
+
+    /** @test */
+    public function itReportsStatisticsCorrectly()
+    {
+        $referenceTypes = [
+            'statisticTotal' => collect(Reference::$REFERENCE_IS_PENDING)->merge(Reference::$REFERENCE_IS_SUBMITTED)->unique()->all(),
+            'statisticRequested' => [Reference::STATUS_REQUESTED],
+            'statisticSubmitted' => Reference::$REFERENCE_IS_SUBMITTED,
+            'statisticUnderReview' => [Reference::STATUS_UNDER_REVIEW],
+            'statisticAccepted' => [Reference::STATUS_ACCEPTED],
+            'statisticRejected' => [Reference::STATUS_REJECTED],
+        ];
+
+        // Check initially zero
+
+        foreach ($referenceTypes as $function => $status) {
+            $this->assertEquals(0, Reference::$function());
+        }
+
+        // Create some references
+
+        factory(Reference::class, 20)->create([
+            'status' => function () use ($referenceTypes) {
+                return Base::randomElement(Base::randomElement($referenceTypes));
+            },
+        ]);
+
+        // Test
```

Comment:
Do we perhaps need to assert that the Cache::remember method is actually called when we try and get it from the model?

Reply:
I mean its not essential for it to cache it - the fact that it is returning the data correctly shows that it works so imo its fine. If you are super keen on it I can put it in though